statocles-site/blog/2024/12/25/jwst/index.markdown

```diff
@@ -0,0 +1,139 @@
+redshift.
+
+Let's start with an image, here is an image of the galaxy:
+![GNz-11 image](https://github.com/user-attachments/assets/52bc0654-8155-4942-9326-eea3be7d7a9b)
+(_Image credit: DAWN JWST Archive_)
+
+This compares the Hubble Space Telescope (HST) and JWST (NIRCam infrared camera) images. It does not look very exciting. But on the right we see the meat of today's demonstration - this shows the location of _spectroscopic slits_. You can think of these are a series of slices through the galaxy where the light is dispersed to make a spectrum. (More info on [how a long-slit spectrograph works](https://en.wikipedia.org/wiki/Long-slit_spectroscopy)).
+
+We can download one of these slit spectra of this object from the DAWN JWST archive:
+
+[https://s3.amazonaws.com/msaexp-nirspec/extractions/goodsn-wide0-v3/goodsn-wide0-v3_prism-clear_1211_1268.spec.fits](https://s3.amazonaws.com/msaexp-nirspec/extractions/goodsn-wide0-v3/goodsn-wide0-v3_prism-clear_1211_1268.spec.fits)
+
+This is in the FITS format which is a RAW-type high dynamic range format used for professional astronomy images.
+
+OK let's make a plot:
+
+    use PDL;
+    use PDL::Graphics::PGPLOT;
+
+    $jwst = rfits 'goodsn-wide0-v3_prism-clear_1211_1268.spec.fits[2]'; # Image is in second FITS extension
+
+    imag $jwst, -0.05, 0.1;
+    @dims = dims($jwst);
+    print join(" ", @dims),"\n";
+
+You can see from the output this is a 2D image of dimensions 435x31 and here is what it looks like:
+
+![2D image](https://github.com/user-attachments/assets/7db5ef7f-e368-4413-89d2-d4aaa76c3246)
+
+The horizontal bright trace down the middle is the spectrum of the galaxy. (The sharp cutoff on the left is due to the extreme redshift of the galaxy as we will see below, so is a genuine astrophysical effect and not an artefact). The dark bands either side are negative traces which _are_ an artefact of the  data reduction pipeline, specifically`sky subtraction via nodding the telescope'. These should be ignored.
+
+## Dimension tricks
+
+Standard procedure would be to sum along the columns of the image to produce a 435 pixel spectrum, summing up the flux of the galaxy along the slit. We could use the `sumover()` function, but `$jwst->sumover` would _sum_ along the rows and produce a 31 pixel ndarray which is not what we want. 'Reduction' operations like `sumover(), average(), medover() `etc. apply along the first axis and make a result with one dimension fewer. To get what we want we have to transpose the image, i.e. swap the two axes. There are various ways of doing this in PDL, the simplest is to use the `mv() `function as in `$jwst->mv(1,0)->sumover()`. This will move dimension 1 to the place of dimension 0, i.e. transposing the 2D ndarray. You can see using `dims() `that the result is a 435 element 1D ndarray.
```

Comment:
Here, the `t` method is the best/shortest way to transpose, not `mv`.

Reply:
Specifically, `t` replaces `mv(1,0)` - `t(1)` will do a complex conjugate, which is not what's sought here. There are docs in `perldl`: `? t`. It's in PDL::Basic, which may not be an ideal location, but that's where `transpose` is, which this is a superior version of (stolen from PDL::LinearAlgebra).

Reply:
now reads as `$jwst->t()->sumover()`